sqlite/comdb2build.c

```diff
@@ -55,15 +55,45 @@ int readIntFromToken(Token* t, int *rst)
     return 1;
 }
 
-static inline int isRemote(Token *t, Token *t2)
+/*
+  Check whether a remote schema name is specified.
+
+  If the schema name is specified and is either "main" or local
+  schema, we copy it to the first argument and return 0.
+*/
+static inline int isRemote(Parse *pParse, Token **t1, Token **t2)
 {
-    return t->n > 0 && (t2 != NULL && t2->n > 0) ;
+    /* Check if the second token is empty. */
+    if ((*t2)->n == 0) {
+        /* No schema name. */
+        return 0;
+    } else {
+        /* We must have both the tokens set. */
+        assert((*t1)->n > 0 && (*t2)->n > 0);
+
+        /* t1 must be a local schema name. */
+        if ((strncasecmp((*t1)->z, thedb->envname, (*t1)->n) == 0) ||
```

Comment:
@dorinhogea : The format used in SQL is either
```
(1) <schema>.<table|index> or
(2) <table|index>
```
In case (1), token t1 holds the schema name while in case (2), the same token t1 holds the table or index name. There's a check at the top of this function that detects whether schema name is supplied. If it is, then token t1 will always have the schema name.
```
static inline int isRemote(Parse *pParse, Token **t1, Token **t2)
{
    /* Check if the second token is empty. */
    if ((*t2)->n == 0) {
        /* No schema name. */
        return 0;
    } else {
```